README.md

```diff
@@ -1,2 +1,307 @@
+
 content creation around image building for containers
+
+# Prerequisites
+
+Before you get started you'll need an IBM Cloud account and need to
+install
+the
+[IBM Cloud Development Tools](https://github.com/IBM-Cloud/ibm-cloud-developer-tools).
+
+You should also clone
+the [guestbook repository](https://github.com/IBM/guestbook) as all
+image builds will be based on that.
+
+Once those are installed, you'll be able to complete the commands
+below.
+
+# Basics of Image Building
+
+Kubernetes uses standard container images to deploy applications.
+
+From the application's perspective, a container image is a single
+filesystem that is created before the application is started.
+
+From a creation perspective, container images are layers. These layers
+are built up using a Docker file. Lets look at the example guestbook
+application Docker file and pull it apart.
+
+```
+FROM golang as builder
+RUN go get github.com/codegangsta/negroni
+RUN go get github.com/gorilla/mux github.com/xyproto/simpleredis
+COPY main.go .
+RUN go build main.go
+
+FROM busybox:ubuntu-14.04
+
+COPY --from=builder /go//main /app/guestbook
+
+ADD public/index.html /app/public/index.html
+ADD public/script.js /app/public/script.js
+ADD public/style.css /app/public/style.css
+
+WORKDIR /app
+CMD ["./guestbook"]
+EXPOSE 3000
+```
+
+Docker files specific a set of commands, at each command we build a
+new layer.
+
+FROM - pulls a layer from a remote repository. The default
+repository is the docker image registry. This could be a complete OS
+image, or some partial image that used to do something small. In this
+example we pull in 2 different base images. One is a named layer for
+the golang build environment, so that it can be referenced later. The
+second one is the base image with just enough operating system to
+provide debug facilities inside the image.
+
+RUN - executes a command, all output files go into the layer. There
+are a few `go get` commands that pull in libraries for the main
+application. Later the main.go is compiled using this.
+
+COPY - copy a file from the local filesystem into the container. This
+can work on a set of files.
+
+WORKDIR - set the working directory for the entrypoint
+
+CMD - what command should be run in this container if no exec is
+passed into it. CMD means that the command run is replaceable. There
+is another stanza `ENTRYPOINT` which is not replaceable.
+
+EXPOSE - open up a specific port from this container to the
+infrastructure. Without this no network traffic will be allowed into
+the container.
+
+These commands, plus many more are documented on the [Docker reference
+site](https://docs.docker.com/engine/reference/builder/#cmd).
+
+## Best Practices for Building Images
+
+There are a number of best practices when building images.
+
+### Clean up everything you can
+
+Whatever files are created in a layer are stored in the intermediate
+layer. Sometimes what gets added in each layer ends up as a
```

Comment:
we should talk about these "layers" too.

Reply:
I guess the question becomes how much of the details to go into here because if this is new to them the optimization part is less interesting then just the how things get done.

Reply:
yea, its tough because I do think this is a very important concept when thinking about the cool features that Docker offers, but given this is "docker build" and not "general docker" it might feel odd to mention it here. However,  you do mention layers and I'm not sure general docker users would ever think about them so if we don't mention them here (since this is where layers are created) I'm not sure where else we'd mention it. 

One of the things I'd like to do is have the main README pretty streamlined, so that people don't get lost in too much text. However, it would also be good to have link to secondary (short) docs (in the same repo) that provide more insight for those who are interested. Perhaps this would be a good topic for one of those secondary docs?